src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -110,44 +121,126 @@ protected override void Dispose(bool disposing)
 
         #region Private methods
 
-        private void StartRequest(RequestCompletionSource state)
+        private async void StartRequest(RequestCompletionSource state)
```

Comment:
The usage pattern of curl_multi_perform is:

curl_multi_perform; // start the request
// Do other work

while(no_more_fds)
{
          curl_multi_perform(); // check which fds have been set
           //  use the data from set fds
}